scrapper.py

```diff
@@ -31,67 +53,226 @@ class Crawler:
         """
-        pass
+        yield self.seed_urls
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler
+    """
+    def get_search_urls(self) -> Iterator[str]:
+        if os.path.exists(CRAWLER_INTERIM):
+            self.seed_urls[0], self.urls, processed = self.load_interim()
+
+            if processed is not None:
+                log.info('All seed_urls are processed.')
+                return
+
+        try:
+
+            while True:
+                response = requests.get(self.seed_urls[0], headers=HEADERS, cookies=COOKIES)
+                response.raise_for_status()
+
+                soup = BeautifulSoup(response.text, 'lxml')
+                next_page = soup.select_one('ul.pagination li:has(span.current) + li a')
+
+                self.seed_urls.append(next_page['href'])
+                yield self.seed_urls.pop(0)
+
+        except RequestException as exc:
+            log.exception("%s was encountered while getting seed_urls", exc)
+            return
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
+    article: Article
+    full_url: str
+    article_id: int
+
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.article = Article(full_url, article_id)
+
+    def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
+        self.article.text = '\n'.join([p.text for p in article_soup.select('.entry-content > p')])
 
-    def _fill_article_with_text(self, article_soup):
-        pass
+    def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
+        self.article.title = article_soup.select_one('h1').text
+        self.article.author = article_soup.select_one('li.autor').text
 
-    def _fill_article_with_meta_information(self, article_soup):
-        pass
+        date: str = article_soup.select_one('time.entry-date').attrs['datetime']
+        self.article.date = self.unify_date_format(date)
+
+        self.article.topics = [tag.text for tag in article_soup.select('a[rel^="category"]')]
 
     @staticmethod
-    def unify_date_format(date_str):
+    def unify_date_format(date_str: str) -> datetime:
         """
         Unifies date format
         """
-        pass
+        return datetime.fromisoformat(date_str)
 
-    def parse(self):
+    def parse(self) -> None:
         """
         Parses each article
         """
-        pass
+        try:
+            response = requests.get(self.full_url, headers=HEADERS, cookies=COOKIES)
+            response.raise_for_status()
+        except RequestException as exc:
+            log.exception("%s was encountered while parsing the article", exc)
+            return
+
+        soup = BeautifulSoup(response.text, features='lxml')
+
+        self._fill_article_with_text(soup)
+        self._fill_article_with_meta_information(soup)
 
+        self.article.save_raw()
```

Comment:
remember S in SOLID? is it responsibility of **parse** method to **save** the file?

Reply:
addressed this issue